canister/src/next_blocks.rs

```diff
@@ -0,0 +1,222 @@
+use crate::types::BlockHash;
+use ic_btc_types::Height;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct NextBlocks {
+    hash_to_height: BTreeMap<BlockHash, Height>,
+    height_to_hash: BTreeMap<Height, Vec<BlockHash>>,
+}
+
+impl NextBlocks {
+    pub(crate) fn insert(&mut self, block_hash: &BlockHash, height: Height) {
```

Comment:
nit: you don't need the `(crate)` for these methods, because this whole module is private to the crate. Similarly for other methods here.